statviz/src/components/custom-graphs/BarChartVertical.tsx

```diff
@@ -0,0 +1,177 @@
+import { Grid } from "@visx/grid";
+import { labelProps, tickProps, tooltipStyles } from "../../utils/chart";
+import { Group } from "@visx/group";
+import { scaleLinear } from "@visx/scale";
+import { AxisBottom, AxisLeft } from "@visx/axis";
+import { scaleBand } from "d3-scale";
+import { Bar } from "@visx/shape";
+import { useTooltip, Tooltip } from "@visx/tooltip";
+import { localPoint } from "@visx/event";
+
+type TooltipData = string;
+
+const marginTop = 20;
+const marginLeft = 70;
+const marginRight = 40;
+const marginBottom = 80;
+
+const defaultSettings = {
+  yEndMargin: 0,
+  hideZeroY: true,
+  hideZeroX: false,
+  startWithZero: true,
+};
+
+export interface IBarChartVertical {
+  fields: {
+    width: number;
+    height: number;
+    background: string;
+    colorBar: string;
+    labelX: string;
+    labelY: string;
+    data: Array<{ x: string; y: number }>;
+    settings?: {
+      yEndMargin?: number;
+      startWithZero?: boolean;
+      hideZeroY?: boolean;
+      hideZeroX?: boolean;
+    };
+  };
+}
+
+export default function BarChartVertical(chart: IBarChartVertical) {
+  const fields = chart.fields;
+
+  const {
+    showTooltip,
+    hideTooltip,
+    tooltipOpen,
+    tooltipData,
+    tooltipLeft = 0,
+    tooltipTop = 0,
+  } = useTooltip<TooltipData>({
+    // initial tooltip state
+    tooltipOpen: false,
+    tooltipLeft: fields.width / 3,
+    tooltipTop: fields.height / 3,
+    tooltipData: "",
+  });
+
+  let tooltipTimeout: number;
+
+  if (!fields.settings) {
+    fields.settings = {};
+  }
+  const settings = {
+    ...defaultSettings,
+    ...fields.settings,
+  };
+
+  const chartWidth = fields.width - (marginLeft + marginRight);
+  const chartHeight = fields.height - (marginTop + marginBottom);
+
+  const y = fields.data.map((e) => e.y);
+  const x = fields.data.map((e) => e.x);
+
+  let yMin = 0;
```

Comment:
in a case like this, I would rather write
`const yMin = !settings.startWithZero ? Math.min(...y) : 0;`

instead of using let

Reply:
@MaikNeubert ?